mlir/include/mlir/Dialect/Bufferization/Transforms/Passes.h

```diff
@@ -166,6 +166,10 @@ struct BufferResultsToOutParamsOpts {
   /// If true, the pass adds a "bufferize.result" attribute to each output
   /// parameter.
   bool addResultAttribute = false;
+  
+  /// If true, the pass eliminates the memref.alloc and memcpy if the returned
+  /// memref is allocated in the current function.
+  bool eliminateAllocCopy = false;
```

Comment:
Maybe `eliminateResultCopy` is more appropriate?

Reply:
It eliminates the allocation as well. :)

Reply:
The allocation is eliminated because it's dead code. The key is eliminate result copy. Also there's another allocation in the caller, `eliminateAllocCopy` is a bit confusing.

Reply:
`eliminate` usually uses for removing something from existing IR. But result copy doesn't exist before this pass, the actual behaviour of this option is to prevent generating extra copy. Thus suggest to rename as `avoidResultCopy`.

Reply:
Besides avoid copying, it also replaces the uses of the allocated memref with the memref in args. `avoidResultCopy` may surprise the user that it also replaces the memref and removes the AllocOp.

Reply:
replace the allocated memref by the memref in arg list is quite straight-forward from the pass name itself, if we have to emphasize removing the alloc and extra copy, how about `avoidBufferResultAllocAndCopy` ?